test/ava-files.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+var test = require('tap').test;
+var AvaFiles = require('../lib/ava-files');
+
+test('testMatcher', function (t) {
+	var avaFiles = new AvaFiles(['**/foo*']);
+
+	var matcher = avaFiles.makeTestMatcher();
+
+	function isTest(file) {
+		t.true(matcher(file), file + ' should be a test');
+	}
+
+	function notTest(file) {
+		t.false(matcher(file), file + ' should not be a test');
+	}
+
+	isTest('foo-bar.js');
+	isTest('foo.js');
+	isTest('foo/blah.js');
+	isTest('bar/foo.js');
+	notTest('bar.js');
+	notTest('bar/bar.js');
+	notTest('_foo-bar.js');
+	notTest('foo/_foo-bar.js');
+	notTest('foo-bar.txt');
+	notTest('node_modules/foo.js');
+	notTest('fixtures/foo.js');
+	notTest('helpers/foo.js');
+	t.end();
+});
+
+test('sourceMatcher', function (t) {
+	var avaFiles = new AvaFiles(['**/foo*']);
+
+	var matcher = avaFiles.makeSourceMatcher();
+
+	function isSource(file) {
+		t.true(matcher(file), file + ' should be a source');
+	}
+
+	function notSource(file) {
+		t.false(matcher(file), file + ' should not be a source');
+	}
+
+	// TODO: Should these actually not be considered sources since they would match as a test?
```

Comment:
It's a trick question. Tests aren't sources, but currently the watcher deems everything that's not a test to be a source. It assumes Chokidar watches the union of tests and sources, and no more.

`isSource` is used to filter test file dependencies.